src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1131,6 +1134,74 @@ public static Collection<FunctionMetadata> ReadFunctionMetadata(ScriptHostConfig
 
+        public static Collection<FunctionMetadata> ReadProxyMetadata(ScriptHostConfiguration config, out object proxyClient, IProxyClientGenerator proxyClientGenerator = null, ScriptSettingsManager settingsManager = null)
+        {
+            proxyClient = null;
+
+            // read the proxy config
+            string proxyConfigPath = Path.Combine(config.RootScriptPath, ScriptConstants.ProxyMetadataFileName);
+            if (!File.Exists(proxyConfigPath))
+            {
+                return null;
+            }
+
+            settingsManager = settingsManager ?? ScriptSettingsManager.Instance;
+
+            var proxyAppSettingValue = settingsManager.GetSetting(EnvironmentSettingNames.ProxySiteExtensionEnabledKey);
+
+            // This is for backward compatibility only, if the file is present but the value of proxy app setting(ROUTING_EXTENSION_VERSION) is explicitly set to 'disabled' we will ignore loading the proxies.
+            if (!string.IsNullOrWhiteSpace(proxyAppSettingValue) && proxyAppSettingValue.Equals("disabled", StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            var proxies = new Collection<FunctionMetadata>();
+            string proxiesJson = File.ReadAllText(proxyConfigPath);
+
+            LoadProxyRoutes(config, proxies, out proxyClient, proxiesJson, proxyClientGenerator);
+
+            return proxies;
+        }
+
+        [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
+        private static void LoadProxyRoutes(ScriptHostConfiguration config, Collection<FunctionMetadata> proxies, out object proxyClient, string proxiesJson, IProxyClientGenerator proxyClientGenerator)
+        {
+            ILoggerFactory loggerFactory = config.HostConfig.LoggerFactory;
+            ILogger proxyStartupLogger = loggerFactory.CreateLogger("Host.Proxies.Initialization");
+
+            if (proxyClientGenerator == null)
+            {
+                proxyClientGenerator = new ProxyClientGenerator();
+            }
+
+            proxyClient = proxyClientGenerator.CreateProxyClient(proxiesJson, proxyStartupLogger);
+
+            if (proxyClient == null)
+            {
+                return;
+            }
+
+            var routes = ((AppService.Proxy.Client.Contract.IProxyClient)proxyClient).GetProxyData();
+
+            foreach (var route in routes.Routes)
+            {
+                var proxyMetadata = new ProxyMetadata();
+
+                // This is just to add a http trigger type for proxies so the pipeline doesn't break, the values are all ignored by Proxy.
+                BindingMetadata bindingMetadata = BindingMetadata.Create(JObject.Parse("{\"authLevel\": \"anonymous\",\"name\": \"req\",\"type\": \"httptrigger\",\"direction\": \"in\"}"));
+
+                proxyMetadata.Bindings.Add(bindingMetadata);
+
+                proxyMetadata.Name = route.Name + "_" + route.Id.ToString();
```

Comment:
Changes made to have at LogFiles/Functions/Proxy/<ProxyName>. 

---
In reply to: [132542984](https://github.com/Azure/azure-webjobs-sdk-script/pull/1749#discussion_r132542984) [](ancestors = 132542984)